src/nuclide_header.F90

```diff
@@ -0,0 +1,822 @@
+module nuclide_header
+
+  use, intrinsic :: ISO_FORTRAN_ENV
+
+  use ace_header
+  use constants
+  use endf,        only: reaction_name
+  use list_header, only: ListInt
+  use math,        only: evaluate_legendre
+  use string
+
+  implicit none
+
+!===============================================================================
+! NUCLIDE_BASE contains the base nuclidic data for a nuclide, which does not depend
+! upon how the nuclear data is represented (i.e., CE, or any variant of MG).
+! The extended types, Nuclide_CE and Nuclide_MG deal with the rest
+!===============================================================================
+
+  type, abstract :: Nuclide_Base
```

Comment:
Also just noticed: `Nuclide_Base`, `Nuclide_CE` etc. should be `NuclideBase` and `NuclideCE` without the underscores